src/CacheInterface.php

```diff
@@ -4,31 +4,40 @@
 
 use React\Promise\PromiseInterface;
 
+/**
+ * All methods only reject with an exception when an error occurs on the underlying storage
+ * layer, for example the connection with Redis dropped and cannot be recovered, or the
+ * directory on the filesystem used to store cache has the wrong permissions or is full.
+ */
 interface CacheInterface
 {
     /**
      * Retrieve an item from the cache, resolves with its value on
-     * success or null when no item can be found.
+     * success or null when no item can be found. It will reject with an exception
+     * on error. (Note that a cache miss isn't an error.)
```

Comment:
Rejecting would be the async equivalent of an `Exception` and I don't see this in PSR-16?